rtrlib/rtr/packets.c

```diff
@@ -1049,6 +1115,56 @@ static int rtr_update_spki_table(struct rtr_socket *rtr_socket, struct spki_tabl
 	return RTR_SUCCESS;
 }
 
+static int rtr_update_aspa_table(struct rtr_socket *rtr_socket, struct aspa_table *aspa_table, const void *pdu)
+{
+	const enum pdu_type type = rtr_get_pdu_type(pdu);
+
+	assert(type == ASPA);
+
+	struct aspa_record record;
+	
+	size_t pdu_size = sizeof(struct pdu_aspa) + ((struct pdu_aspa *)pdu)->provider_count * sizeof(((struct pdu_aspa *)pdu)->provider_asns[0]);
+
+	rtr_aspa_pdu_2_aspa_record(rtr_socket, pdu, &record, type);
+
+	int rtval;
+	if (((struct pdu_aspa *)pdu)->flags == 1) {
+		rtval = aspa_table_add(aspa_table, &record, rtr_socket);
+
+	} else if (((struct pdu_aspa *)pdu)->flags == 0) {
+		rtval = aspa_table_remove(aspa_table, &record, rtr_socket);
+
+	} else {
+		const char txt[] = "ASPA PDU with invalid flags value received";
+
+		RTR_DBG("%s", txt);
```

Comment:
why not inline the string ?

Reply:
its used in the pdu below again -- again a case where I just copied code from the pfx and spki cases